include/pybind11/pytypes.h

```diff
@@ -375,31 +388,101 @@ PYBIND11_NAMESPACE_END(detail)
 /// python).
 class PYBIND11_EXPORT_EXCEPTION error_already_set : public std::runtime_error {
 public:
-    /// Constructs a new exception from the current Python error indicator, if any.  The current
-    /// Python error indicator will be cleared.
-    error_already_set() : std::runtime_error(detail::error_string()) {
+    /// Fetches the current Python exception (using PyErr_Fetch()), which will clear the
+    /// current Python error indicator.
+    error_already_set() : std::runtime_error("") {
+        if (!PyErr_Occurred()) {
```

Comment:
Don't bother doing PyErr_Occurred here, it is unnecessary overhead. Just check if the m_type has a nullptr after PyErr_Fetch, it should be a lot faster and more consistent with how we handle pytypes.